src/components/PrimaryPanes/SourcesTree.js

```diff
@@ -115,17 +129,19 @@ class SourcesTree extends Component<Props, State> {
     }
 
     if (nextProps.shownSource && nextProps.shownSource != shownSource) {
-      const matchingSources = Object.keys(sources).filter(sourceId => {
-        return getRawSourceURL(sources[sourceId].url) === nextProps.shownSource;
-      });
+      const matchingRawSource = getRawSourceByUrl(
+        sources,
+        nextProps.shownSource
+      );
+      const matchingSelectionSource = getSourceByUrl(
+        sources,
+        nextProps.shownSource
+      );
 
-      if (matchingSources.length) {
-        const listItems = getDirectories(
-          sources[matchingSources[0]],
-          sourceTree
-        );
-        if (listItems && listItems.length) {
-          this.selectItem(listItems[0]);
+      if (matchingRawSource) {
+        const listItems = getDirectories(matchingRawSource, sourceTree);
+        if (matchingSelectionSource) {
+          this.props.selectSource(matchingSelectionSource.id);
```

Comment:
Calling `this.props.selectSource` directly instead of `setItem` because `setItem` requires an item as an argument, and there will never be an item for pretty sources